app/documents.js

```diff
@@ -172,7 +173,11 @@ export class Document extends Base {
       const doc = await this.buildDocument(model, i);
       update();
       await delay(0);
-      this.documents[model.name].push(doc);
+      // only push the document if the key is not already taken
+      if (built_docs_map[doc.__key] === undefined) {
+        built_docs_map[doc.__key] = doc;
+        this.documents[model.name].push(doc);
+      }
```

Comment:
After line 180 add the following code. Resting `build_docs_map` to be an empty object will free up memory to be used elsewhere.

```js
.then(() => {
  built_docs_map = {};
})
```